applications/PfemFluidDynamicsApplication/custom_elements/three_step_updated_lagrangian_element.cpp

```diff
@@ -0,0 +1,1643 @@
+      noalias(AccelerationValues) += VelocityValues / TimeStep; //these are negative accelerations
+      noalias(rRightHandSideVector) += prod(MassMatrix, AccelerationValues);
+      noalias(rLeftHandSideMatrix) += MassMatrix / TimeStep;
+    }
+
+    // // using BDF coefficients
+    // const Vector &rBDFCoeffs = rCurrentProcessInfo[BDF_COEFFICIENTS];
+    // noalias(rLeftHandSideMatrix) += rBDFCoeffs[0] * MassMatrix;
+    // VectorType TimeTerm = rBDFCoeffs[0] * LastValues;
+    // for (SizeType i = 1; i < rBDFCoeffs.size(); i++)
+    // {
+    //   this->GetVelocityValues(LastValues, i);
+    //   noalias(TimeTerm) += rBDFCoeffs[i] * LastValues;
+    // }
+    // noalias(rRightHandSideVector) -= prod(MassMatrix, TimeTerm);
+  }
+
+  // this is a part of the last iteration of the fractional step
+  template <unsigned int TDim>
+  void ThreeStepUpdatedLagrangianElement<TDim>::Calculate(const Variable<array_1d<double, 3>> &rVariable,
+                                                          array_1d<double, 3> &rOutput,
+                                                          const ProcessInfo &rCurrentProcessInfo)
+  {
+    if (rVariable == VELOCITY)
+    {
+      GeometryType &rGeom = this->GetGeometry();
+      const SizeType NumNodes = rGeom.PointsNumber();
+      const SizeType LocalSize = TDim * NumNodes;
+
+      // Shape functions and integration points
+      ShapeFunctionDerivativesArrayType DN_DX;
+      Matrix NContainer;
+      VectorType GaussWeights;
+      this->CalculateGeometryData(DN_DX, NContainer, GaussWeights);
+      const unsigned int NumGauss = GaussWeights.size();
+
+      VectorType NodalVelCorrection = ZeroVector(LocalSize);
+
+      // Loop on integration points
+      for (unsigned int g = 0; g < NumGauss; ++g)
+      {
+        const ShapeFunctionsType &N = row(NContainer, g);
+        const ShapeFunctionDerivativesType &rDN_DX = DN_DX[g];
+
+        double Density = 0;
+        this->EvaluateInPoint(Density, DENSITY, N);
+
+        const double TimeStep = rCurrentProcessInfo[DELTA_TIME];
+        // double timeFactor = 1.0 / rCurrentProcessInfo[BDF_COEFFICIENTS][0];
+        // double timeFactor = 0.5 * TimeStep; //second order accelerations
+        double timeFactor = TimeStep; //first order accelerations
+        const double Coeff = GaussWeights[g] * timeFactor / Density;
+
+        // Calculate contribution to the gradient term (RHS)
+        // double DeltaPressure;
+        // this->EvaluateInPoint(DeltaPressure, PRESSURE_OLD_IT, N);
+
+        const unsigned int schemeOrder = 2;
+        double elementalPressure = 0;
+        if (schemeOrder == 2)
+        {
+          this->EvaluateDifferenceInPoint(elementalPressure, PRESSURE, N);
+        }
+        else
+        {
+          this->EvaluateInPoint(elementalPressure, PRESSURE, N);
+        }
+
+        SizeType RowIndex = 0;
+
+        for (SizeType i = 0; i < NumNodes; ++i)
+        {
+          for (SizeType d = 0; d < TDim; ++d)

```

Comment:
the for's should be:
~~~c++
for (----) {
    (code)
}
~~~
According to style